### ISSUE TITLE
Wrong import of bank statement file for multiple accounts

### ISSUE THREAD

**Issue by sbidoul** [odoo/odoo#33162]
Impacted versions: 12.0, 11.0, 10.0

The current `bank_statement_import` `_parse_file()` api can return multiple statements but all for the same account number.

Some bank statement files (eg CAMT) can contain statements for multiple accounts.

It is therefore impossible to correctly import such bank statements, and some parsers incorrectly return the account number of the last statement found in the file.

**Comment by gdgellatly** [comment]
@sbidoul I actually found this a couple of weeks ago, because the primary NZ format has multiple accounts in one file as well. However my problem was a little bit more difficult because statements were multiple companies as well.

What we ended up doing was creating a second import frontend, putting some security around that, and then overriding the import button to call the new frontend, which split the files into 1 per account and then creates 1 instance of the wizard per file and imports.

Only in production a couple of weeks but seems to run well.

**Comment by sbidoul** [comment]
OPW 1999593

**Comment by pedrobaeza** [comment]
Coming to this through codetriage.com. Any answer from OPW?

**Comment by sbidoul** [comment]
@pedrobaeza in enterprise 12.0, an error message is now raised if a CAMT file contains statement for different bank accounts.

**Comment by pedrobaeza** [comment]
Should we close this then?

**Comment by sbidoul** [comment]
No, IMO it is still a valid feature request to support multi-account statement files.
